### PR TITLE
Update docstring for tf.reduce_sum/reduce.

### LINKED ISSUE #19498: tf.reduce_sum not consistent with documentation when axis==[ ]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7.0
- **Python version**:  3.5.2
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:  9.0.176 / 7.0.5
- **GPU model and memory**: GeForce GTX 1050 Ti/PCIe/SSE2 4GB
- **Exact command to reproduce**: custom script

### Describe the problem
documentation says:
If axis has no entries, all dimensions are reduced, and a tensor with a single element is returned.

In practice:
If axis == []: nothing is reduced (the output is the input itself)

### Source code / logs
import tensorflow as tf

constant = tf.constant([[1., 2.], [3., 4.]])
reduce = tf.reduce_sum(constant, axis=[])

with tf.Session() as sess:
    print(sess.run(reduce))

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #19498 where the description for reduction:
```
If axis has no entries, all dimensions are reduced, and a tensor with a single element is returned.
```

does not match the current behavior. The current behavior (matches `np.sum` as well) is actually:
```
If axis is None, all dimensions are reduced, and a tensor with a single element is returned.
```

This fix fixes the discrepancy for tf.reduce_sum/mean/max/min/prod.

This fix fixes #19498.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by case540 — APPROVED** [review]

**Comment by case540** [comment]
Thanks for the doc update!
